core/trino-main/src/main/java/io/trino/operator/aggregation/AggregatorFactory.java

```diff
@@ -49,12 +52,21 @@ public AggregatorFactory(
         this.step = requireNonNull(step, "step is null");
         this.intermediateType = requireNonNull(intermediateType, "intermediateType is null");
         this.finalType = requireNonNull(finalType, "finalType is null");
-        this.inputChannels = ImmutableList.copyOf(requireNonNull(inputChannels, "inputChannels is null"));
+        requireNonNull(inputChannels, "inputChannels is null");
+        if (step.isInputRaw()) {
+            intermediateStateChannel = OptionalInt.empty();
+            this.rawInputChannels = ImmutableList.copyOf(inputChannels);
+        }
+        else {
+            intermediateStateChannel = OptionalInt.of(inputChannels.get(0));
```

Comment:
Not really, what we talk about here is arguments to the aggregation function. The list of arguments is translated from symbols to channel indexes but still, the first argument is the intermediate state for final aggregation. so the index 0 in this list is the "magic" number. In order to avoid that we would have to have a separate field for the intermediate state symbol in the Aggregation class (the `intermediateSymbol` local var in the `io.trino.sql.planner.iterative.rule.PushPartialAggregationThroughExchange#split`). That may be a good idea anyway but it's not related to this change.